### ISSUE TITLE
Easier way to construct NetSdkManagedInstallationRecordRepository 

### ISSUE THREAD

**Issue by wli3** [dotnet/sdk#18021]
To answer what workload is installed on the machine, currently you need to get a `WorkloadInstaller` from `WorkloadInstallerFactory`, which need a lot of dependency while NetSdkManagedInstallationRecordRepository  only need dotnet sdk path. There should be an easy way to construct like NetSdkManagedInstallationRecordRepositoryFactory. And that will produce the right object on admin install or SDK managed install.

```c#
NetSdkManagedInstallationRecordRepository from WorkloadInstallerFactory
    .GetWorkloadInstaller(_reporter, _sdkFeatureBand, workloadResolver, _verbosity)
    .GetWorkloadInstallationRecordRepository();

```

**Comment by sfoslund** [comment]
Note: I think we'll have to create some sort of factory for installation record repositories eventually anyways since the resolver will need to use them for reading installation records. We'll probably have to move the logic @joeloff is implementing to determine which type of installation we're using into https://github.com/dotnet/sdk/tree/main/src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallRecords, since that dir is shared by the resolver.

**Comment by joeloff** [comment]
Yes, for admin installs, the construction of the repository instance is more complicated. Because the factory can write (or delete) items from the registry, it requires an elevation context to know A) whether it needs to elevate and B) whether the process was already elevated because of a previously privileged action, e.g. you installed something and now want to write a record, at which point the install would have elevated. If elevation is needed there's a large  set of actions that need to be executed. I can sync up with you tomorrow to go over some details

**Comment by sfoslund** [comment]
We'll need to address this once the admin install PR is merged, since there's at least one place where we use the file based install method directly, which will not work in admin scenarios: https://github.com/dotnet/sdk/blob/main/src/Cli/dotnet/commands/dotnet-workload/list/WorkloadListCommand.cs#L60

**Comment by joeloff** [comment]
My next round of admin fixes will partially address this (make it 'list' work with admin installs). Still won't fully simplify it
